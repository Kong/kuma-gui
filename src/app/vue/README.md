---
section: Packages
---
# vue

Automatic `vue` initialization using our projects dependency injection layer.

