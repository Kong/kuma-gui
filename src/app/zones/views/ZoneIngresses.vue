<template>
  <div class="zoneingresses">
    <MultizoneInfo v-if="multicluster === false" />

    <!-- Zone CPs information for when Multicluster is enabled -->
    <FrameSkeleton v-else>
      <DataOverview
        :selected-entity-name="entity?.name"
        :page-size="pageSize"
        :is-loading="isLoading"
        :error="error"
        :empty-state="empty_state"
        :table-data="tableData"
        :table-data-is-empty="isEmpty"
        :next="next"
        :page-offset="pageOffset"
        @table-action="tableAction"
        @load-data="loadData"
      >
        <template #additionalControls>
          <KButton
            v-if="$route.query.ns"
            class="back-button"
            appearance="primary"
            icon="arrowLeft"
            :to="{ name: 'zoneingresses' }"
          >
            View all
          </KButton>
        </template>
      </DataOverview>
      <TabsWidget
        v-if="isEmpty === false"
        :has-error="error !== null"
        :is-loading="isLoading"
        :tabs="tabs"
        initial-tab-override="overview"
      >
        <template #tabHeader>
          <h1 class="entity-heading">
            Zone Ingress: {{ entity.name }}
          </h1>
        </template>
        <template #overview>
          <LabelList>
            <div>
              <ul>
                <li
                  v-for="(value, key) in entity"
                  :key="key"
                >
                  <h4>
                    {{ key }}
                  </h4>
                  <p>
                    {{ value }}
                  </p>
                </li>
              </ul>
            </div>
          </LabelList>
        </template>
        <template #insights>
          <KCard border-variant="noBorder">
            <template #body>
              <AccordionList :initially-open="0">
                <AccordionItem
                  v-for="(value, key) in subscriptionsReversed"
                  :key="key"
                >
                  <template #accordion-header>
                    <SubscriptionHeader :details="value" />
                  </template>

                  <template #accordion-content>
                    <SubscriptionDetails
                      :details="value"
                      is-discovery-subscription
                    />
                  </template>
                </AccordionItem>
              </AccordionList>
            </template>
          </KCard>
        </template>

        <template #xds-configuration>
          <EnvoyData
            data-path="xds"
            :zone-ingress-name="entity.name"
            query-key="envoy-data-zone-ingress"
          />
        </template>

        <template #envoy-stats>
          <EnvoyData
            data-path="stats"
            :zone-ingress-name="entity.name"
            query-key="envoy-data-zone-ingress"
          />
        </template>

        <template #envoy-clusters>
          <EnvoyData
            data-path="clusters"
            :zone-ingress-name="entity.name"
            query-key="envoy-data-zone-ingress"
          />
        </template>
      </TabsWidget>
    </FrameSkeleton>
  </div>
</template>

<script>
import { mapGetters } from 'vuex'
import { KButton, KCard } from '@kong/kongponents'

import { getItemStatusFromInsight } from '@/utilities/dataplane'
import { getSome } from '@/utilities/helpers'
import { kumaApi } from '@/api/kumaApi'
import { PAGE_SIZE_DEFAULT } from '@/constants'
import { QueryParameter } from '@/utilities/QueryParameter'
import AccordionItem from '@/app/common/AccordionItem.vue'
import AccordionList from '@/app/common/AccordionList.vue'
import DataOverview from '@/app/common/DataOverview.vue'
import EnvoyData from '@/app/common/EnvoyData.vue'
import FrameSkeleton from '@/app/common/FrameSkeleton.vue'
import LabelList from '@/app/common/LabelList.vue'
import MultizoneInfo from '../components/MultizoneInfo.vue'
import SubscriptionDetails from '@/app/common/subscriptions/SubscriptionDetails.vue'
import SubscriptionHeader from '@/app/common/subscriptions/SubscriptionHeader.vue'
import TabsWidget from '@/app/common/TabsWidget.vue'

export default {
  name: 'ZoneIngresses',

  components: {
    AccordionItem,
    AccordionList,
    DataOverview,
    EnvoyData,
    FrameSkeleton,
    LabelList,
    MultizoneInfo,
    SubscriptionDetails,
    SubscriptionHeader,
    TabsWidget,
    KButton,
    KCard,
  },

  props: {
    selectedZoneIngressName: {
      type: String,
      required: false,
      default: null,
    },

    offset: {
      type: Number,
      required: false,
      default: 0,
    },
  },

  data() {
    return {
      isLoading: true,
      isEmpty: false,
      error: null,

      empty_state: {
        title: 'No Data',
        message: 'There are no Zone Ingresses present.',
      },
      tableData: {
        headers: [
          { key: 'actions', hideLabel: true },
          { label: 'Status', key: 'status' },
          { label: 'Name', key: 'name' },
        ],
        data: [],
      },
      tabs: [
        {
          hash: '#overview',
          title: 'Overview',
        },
        {
          hash: '#insights',
          title: 'Zone Ingress Insights',
        },
        {
          hash: '#xds-configuration',
          title: 'XDS Configuration',
        },
        {
          hash: '#envoy-stats',
          title: 'Stats',
        },
        {
          hash: '#envoy-clusters',
          title: 'Clusters',
        },
      ],
      entity: {},
      rawData: [],
      pageSize: PAGE_SIZE_DEFAULT,
      next: null,
      subscriptionsReversed: [],
      pageOffset: this.offset,
    }
  },

  computed: {
    ...mapGetters({
      multicluster: 'config/getMulticlusterStatus',
    }),
  },

  watch: {
    $route() {
      // Ensures basic state is reset when switching meshes using the mesh selector.
      this.isLoading = true
      this.isEmpty = false
      this.error = null

      this.init(0)
    },
  },

  beforeMount() {
    this.init(this.offset)
  },

  methods: {
    init(offset) {
      if (this.multicluster) {
        this.loadData(offset)
      }
    },
    tableAction(ev) {
      const data = ev

      // load the data into the tabs
      this.getEntity(data)
    },

    async loadData(offset) {
      this.pageOffset = offset
      // Puts the offset parameter in the URL so it can be retrieved when the user reloads the page.
      QueryParameter.set('offset', offset > 0 ? offset : null)

      this.isLoading = true
      this.isEmpty = false

      const name = this.$route.query.ns || null
      const size = this.pageSize

      try {
        const { data, next } = await this.getZoneIngressOverviews(name, size, offset)

        // set pagination
        this.next = next

        // set table data
        if (data.length) {
          this.isEmpty = false
          this.rawData = data
          this.getEntity({ name: this.selectedZoneIngressName ?? data[0].name })

          this.tableData.data = data.map((item) => {
            const { zoneIngressInsight = {} } = item
            const status = getItemStatusFromInsight(zoneIngressInsight)

            return { ...item, status }
          })
        } else {
          this.tableData.data = []
          this.isEmpty = true
        }
      } catch (err) {
        if (err instanceof Error) {
          this.error = err
        } else {
          console.error(err)
        }

        this.isEmpty = true
      } finally {
        this.isLoading = false
      }
    },

    getEntity(entity) {
      const selected = ['type', 'name']
      const item = this.rawData.find((data) => data.name === entity.name)

      const subscriptions = item?.zoneIngressInsight?.subscriptions ?? []

      this.subscriptionsReversed = Array.from(subscriptions).reverse()

      this.entity = getSome(item, selected)
      QueryParameter.set('zoneIngress', this.entity.name)
    },

    /**
     * @param {string | null} name
     * @param {number} size
     * @param {number} offset
     * @returns {Promise<{ data: ZoneIngressOverview[], next: string | null }>}
     */
    async getZoneIngressOverviews(name, size, offset) {
      if (name) {
        const zoneIngressOverview = await kumaApi.getZoneIngressOverview({ name }, { size, offset })

        return {
          data: [zoneIngressOverview],
          next: null,
        }
      } else {
        const { items, next } = await kumaApi.getAllZoneIngressOverviews({ size, offset })

        return {
          data: items ?? [],
          next,
        }
      }
    },
  },
}
</script>
