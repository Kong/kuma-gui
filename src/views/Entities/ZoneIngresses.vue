<template>
  <div class="zoneingresses">
    <MultizoneInfo v-if="multicluster === false" />

    <!-- Zone CPs information for when Multicluster is enabled -->
    <FrameSkeleton v-else>
      <DataOverview
        :page-size="pageSize"
        :has-error="hasError"
        :is-loading="isLoading"
        :empty-state="empty_state"
        :table-data="tableData"
        :table-data-is-empty="tableDataIsEmpty"
        :show-warnings="tableData.data.some((item) => item.withWarnings)"
        :next="next"
        @tableAction="tableAction"
        @loadData="loadData($event)"
      />
      <Tabs
        v-if="isEmpty === false"
        :has-error="hasError"
        :is-loading="isLoading"
        :tabs="tabs"
        initial-tab-override="overview"
      >
        <template v-slot:tabHeader>
          <h3 v-if="entity">
            Zone Ingress: {{ entity.name }}
          </h3>
        </template>
        <template v-slot:overview>
          <LabelList
            :has-error="entityHasError"
            :is-loading="entityIsLoading"
            :is-empty="entityIsEmpty"
          >
            <div>
              <ul>
                <li
                  v-for="(value, key) in entity"
                  :key="key"
                >
                  <h4 v-if="value">
                    {{ key }}
                  </h4>
                  <p v-if="key === 'status'">
                    <KBadge :appearance="value === 'Offline' ? 'danger' : 'success'">
                      {{ value }}
                    </KBadge>
                  </p>
                  <p v-else>
                    {{ value }}
                  </p>
                </li>
              </ul>
            </div>
          </LabelList>
        </template>
        <template v-slot:insights>
          <KCard border-variant="noBorder">
            <template v-slot:body>
              <Accordion :initially-open="0">
                <AccordionItem
                  v-for="(value, key) in zoneIngressInsightSubscriptionsReversed"
                  :key="key"
                >
                  <template v-slot:accordion-header>
                    <ZoneInsightSubscriptionHeader :details="value" />
                  </template>

                  <template v-slot:accordion-content>
                    <ZoneInsightSubscriptionDetails :details="value" />
                  </template>
                </AccordionItem>
              </Accordion>
            </template>
          </KCard>
        </template>
      </Tabs>
    </FrameSkeleton>
  </div>
</template>

<script>
import get from 'lodash/get'
import { mapGetters } from 'vuex'
import { getTableData } from '@/utils/tableDataUtils'
import { getSome } from '@/helpers'
import Kuma from '@/services/kuma'
import FrameSkeleton from '@/components/Skeletons/FrameSkeleton'
import DataOverview from '@/components/Skeletons/DataOverview'
import Tabs from '@/components/Utils/Tabs'
import LabelList from '@/components/Utils/LabelList'

import { getItemStatusFromInsight } from '@/dataplane'
import { PAGE_SIZE_DEFAULT } from '@/consts'

import Accordion from '@/components/Accordion/Accordion'
import AccordionItem from '@/components/Accordion/AccordionItem'

import ZoneInsightSubscriptionDetails from './components/ZoneInsightSubscriptionDetails'
import ZoneInsightSubscriptionHeader from './components/ZoneInsightSubscriptionHeader'
import MultizoneInfo from './components/MultizoneInfo'

export default {
  name: 'ZoneIngresses',
  components: {
    FrameSkeleton,
    DataOverview,
    Tabs,
    LabelList,
    Accordion,
    AccordionItem,
    ZoneInsightSubscriptionDetails,
    ZoneInsightSubscriptionHeader,
    MultizoneInfo,
  },

  metaInfo: {
    title: 'ZoneIngresses',
  },
  data() {
    return {
      isLoading: true,
      isEmpty: false,
      hasError: false,
      entityIsLoading: true,
      entityIsEmpty: false,
      entityHasError: false,
      tableDataIsEmpty: false,
      empty_state: {
        title: 'No Data',
        message: 'There are no Zone Ingresses present.',
      },
      tableData: {
        headers: [
          { key: 'actions', hideLabel: true },
          { label: 'Status', key: 'status' },
          { label: 'Name', key: 'name' },
        ],
        data: [],
      },
      tabs: [
        {
          hash: '#overview',
          title: 'Overview',
        },
        {
          hash: '#insights',
          title: 'Zone Ingress Insights',
        },
      ],
      entity: {},
      pageSize: PAGE_SIZE_DEFAULT,
      next: null,
      zoneIngressInsightSubscriptionsReversed: [],
    }
  },
  computed: {
    ...mapGetters({
      multicluster: 'config/getMulticlusterStatus',
    }),
    pageTitle() {
      return this.$route.meta.title
    },
  },
  watch: {
    $route() {
      this.init()
    },
  },
  beforeMount() {
    this.init()
  },
  methods: {
    init() {
      if (this.multicluster) {
        this.loadData()
      }
    },
    tableAction(ev) {
      const data = ev

      // load the data into the tabs
      this.getEntity(data)
    },

    async loadData(offset = '0') {
      this.isLoading = true
      this.isEmpty = false

      try {
        const { data, next } = await getTableData({
          getAllEntities: Kuma.getAllZoneIngressOverviews.bind(Kuma),
          size: this.pageSize,
          offset,
        })

        // set pagination
        this.next = next

        // set table data
        if (data.length) {
          this.tableData.data = data.map((item) => {
            const { zoneIngressInsight = {} } = item

            return { ...item, ...getItemStatusFromInsight(zoneIngressInsight) }
          })
          this.tableDataIsEmpty = false
          this.isEmpty = false

          this.getEntity({ name: data[0].name })
        } else {
          this.tableData.data = []
          this.tableDataIsEmpty = true
          this.isEmpty = true
          this.entityIsEmpty = true
        }
      } catch (error) {
        this.hasError = true
        this.isEmpty = true

        console.error(error)
      } finally {
        this.isLoading = false
      }
    },
    async getEntity(entity) {
      this.entityIsLoading = true
      this.entityIsEmpty = true

      const selected = ['type', 'name', 'mesh']

      const timeout = setTimeout(() => {
        this.entityIsEmpty = true
        this.entityIsLoading = false
      }, process.env.VUE_APP_DATA_TIMEOUT)

      if (entity) {
        this.entityIsEmpty = false

        try {
          // get the ZoneIngress details from the ZoneIngress Insights endpoint
          const response = await Kuma.getZoneIngressOverview({ name: entity.name })

          const subscriptions = get(response, 'zoneIngressInsight.subscriptions', [])

          this.zoneIngressInsightSubscriptionsReversed = Array.from(subscriptions).reverse()

          this.entity = getSome(response, selected)
        } catch (e) {
          console.error(e)

          this.entity = null
          this.entityHasError = true
          this.entityIsEmpty = true
        } finally {
          clearTimeout(timeout)
        }
      }

      this.entityIsLoading = false
    },
  },
}
</script>
