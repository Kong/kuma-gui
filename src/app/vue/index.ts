import { createApp } from 'vue'
import {
  createRouter,
  createWebHistory,
} from 'vue-router'

import type { ServiceDefinition } from '@/services/utils'
import { token, createInjections } from '@/services/utils'
import type { Component } from 'vue'
import type { Router, RouteRecordRaw, NavigationGuard } from 'vue-router'

type Token = ReturnType<typeof token>

type VueApp = ReturnType<typeof createApp>
export type PluginDefinition = Parameters<VueApp['use']>
export type ComponentDefinition = [string, Component]

const $ = {
  app: token<(App: Component) => Promise<VueApp>>('vue.app'),
  router: token<Router>('vue.router'),

  components: token('vue.components'),
  plugins: token<PluginDefinition[]>('vue.plugins'),
  routes: token<RouteRecordRaw[]>('vue.routes'),
  routesLabel: token<RouteRecordRaw[]>('vue.routes.label'),
  navigationGuards: token<NavigationGuard[]>('vue.routes.navigation.guards'),
}
export const services = (app: Record<string, Token>): ServiceDefinition[] => {
  return [
    [$.app, {
      service: (
        components: ComponentDefinition[],
        plugins: PluginDefinition[],
      ) => {
        return async (App: Component) => {
          const app = createApp(App)

          plugins.forEach(([...args]) => {
            app.use(...args)
          })

          components.forEach(([name, component]: [string, Component]) => {
            app.component(name, component)
          })

          return app
        }
      },
      arguments: [
        $.components,
        $.plugins,
      ],
    }],
    [$.router, {
      service: (
        env: (str: 'KUMA_BASE_PATH') => string,
        routes: RouteRecordRaw[],
        guards: NavigationGuard[],
      ) => {
        const router = createRouter({
          history: createWebHistory(env('KUMA_BASE_PATH')),
          routes,
        })

        guards.forEach((item) => {
          if (typeof item === 'function') {
            router.beforeEach(item)
          }
        })

        return router
      },
      arguments: [
        app.env,
        $.routes,
        $.navigationGuards,
      ],
    }],

    [token('vue.plugins'), {
      service: (router) => {
        return [
          [router],
        ]
      },
      arguments: [
        $.router,
      ],
      labels: [
        $.plugins,
      ],
    }],

    // TODO(jc): Label decoration. Make it so we can decorate labels
    // Temporarily turn the routesLabel label into routes for decoration
    // purposes
    [$.routes, {
      service: (routes: RouteRecordRaw[]) => routes,
      arguments: [
        $.routesLabel,
      ],
    }],
    //

    // TODO(jc): Make it so we don't need to provide a [] empty default for
    // labels
    [token('application.routes.navigation.guards'), {
      service: () => {
        return []
      },
      labels: [
        $.navigationGuards,
      ],
    }],
  ]
}

export const TOKENS = $
export const [
  useRouter,
] = createInjections(
  TOKENS.router,
)
