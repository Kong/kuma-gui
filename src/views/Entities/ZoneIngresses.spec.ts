import { createStore } from 'vuex'
import { flushPromises, RouterLinkStub } from '@vue/test-utils'
import { render, screen } from '@testing-library/vue'
import userEvent from '@testing-library/user-event'
import { KAlert, KBadge, KButton, KCard, KClipboardProvider, KEmptyState, KIcon, KPop, KTable, KTabs } from '@kong/kongponents'

import ZoneIngresses from './ZoneIngresses.vue'

jest.mock('@/helpers', () => {
  const originalModule = jest.requireActual('@/helpers')

  return {
    __esModule: true,
    ...originalModule,
    humanReadableDate: (data: string) => data,
    rawReadableDate: (data: string) => data,
  }
})

function renderComponent() {
  const store = createStore({
    modules: {
      config: {
        namespaced: true,
        state: {
          clientConfig: {
            mode: 'global',
          },
        },
        getters: {
          getMulticlusterStatus: () => true,
        },
      },
    },
  })

  return render(ZoneIngresses, {
    global: {
      plugins: [store],
      stubs: {
        'router-link': RouterLinkStub,
      },
      components: { KAlert, KBadge, KButton, KCard, KClipboardProvider, KEmptyState, KIcon, KPop, KTable, KTabs },
      mocks: {
        $route: {
          query: {},
        },
      },
    },
  })
}

describe('ZoneIngresses.vue', () => {
  it('renders snapshot when no multizone', async () => {
    const { container } = renderComponent()

    await flushPromises()

    expect(container).toMatchSnapshot()
  })

  it('renders snapshot when multizone', async () => {
    const { container } = renderComponent()

    await flushPromises()

    await screen.findByText(/ZoneIngressOverview/)

    expect(container).toMatchSnapshot()
  })

  it('renders zoneingress insights', async () => {
    renderComponent()

    await screen.findByText(/ZoneIngressOverview/)

    await userEvent.click(screen.getByText(/Zone Ingress Insights/))
    expect(screen.getByTestId('tab-container')).toMatchSnapshot()
  })
})
