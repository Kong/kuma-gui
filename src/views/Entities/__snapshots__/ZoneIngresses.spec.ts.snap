// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ZoneIngresses.vue renders snapshot when multizone 1`] = `
<div>
  <div
    class="zoneingresses"
  >
    <div
      class="component-frame"
    >
      <div
        class="data-overview"
        data-testid="data-overview"
      >
        <div
          class="data-table-controls mb-2"
        >
           
          <button
            class="k-button ml-2 refresh-button small primary"
            data-v-38fa8ecf=""
            type="button"
          >
            <div
              class="refresh-icon"
              data-v-38fa8ecf=""
            >
              <svg
                data-v-38fa8ecf=""
                viewBox="0 0 36 36"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  data-v-38fa8ecf=""
                  fill="#fff"
                  fill-rule="nonzero"
                >
                  <path
                    d="M18 5.5a12.465 12.465 0 00-8.118 2.995 1.5 1.5 0 001.847 2.363l.115-.095A9.437 9.437 0 0118 8.5l.272.004a9.487 9.487 0 019.07 7.75l.04.246H25a.5.5 0 00-.416.777l4 6a.5.5 0 00.832 0l4-6 .04-.072A.5.5 0 0033 16.5h-2.601l-.017-.15C29.567 10.2 24.294 5.5 18 5.5zM2.584 18.723l-.04.072A.5.5 0 003 19.5h2.6l.018.15C6.433 25.8 11.706 30.5 18 30.5c3.013 0 5.873-1.076 8.118-2.995a1.5 1.5 0 00-1.847-2.363l-.115.095A9.437 9.437 0 0118 27.5l-.272-.004a9.487 9.487 0 01-9.07-7.75l-.041-.246H11a.5.5 0 00.416-.777l-4-6a.5.5 0 00-.832 0l-4 6z"
                    data-v-38fa8ecf=""
                  />
                </g>
              </svg>
            </div>
             
            <span
              data-v-38fa8ecf=""
            >
              Refresh
            </span>
          </button>
        </div>
         
        <div
          class="data-overview-content"
        >
          <div
            class="data-overview-table"
          >
            <table
              class="k-table micro-table has-hover is-clickable"
              data-v-bfeabd48=""
            >
              <thead
                data-v-bfeabd48=""
              >
                <tr
                  data-v-bfeabd48=""
                >
                  <th
                    class=""
                    data-v-bfeabd48=""
                  >
                    <!---->
                  </th>
                  <th
                    class=""
                    data-v-bfeabd48=""
                  >
                     Status 
                  </th>
                  <th
                    class=""
                    data-v-bfeabd48=""
                  >
                     Name 
                  </th>
                </tr>
              </thead>
              <tbody
                data-v-bfeabd48=""
              >
                <tr
                  data-v-bfeabd48=""
                >
                  <td
                    data-v-bfeabd48=""
                  >
                    <a
                      class="data-table-action-link is-active"
                      data-v-bfeabd48=""
                    >
                      <span
                        class="action-link__active-state"
                        data-v-bfeabd48=""
                      >
                        
              âœ“
              
                        <span
                          class="sr-only"
                          data-v-bfeabd48=""
                        >
                          
                Selected
              
                        </span>
                      </span>
                    </a>
                  </td>
                  <td
                    data-v-bfeabd48=""
                  >
                    <div
                      class="entity-status"
                      data-v-bfeabd48=""
                    >
                      <span
                        class="entity-status__dot"
                        data-v-bfeabd48=""
                      />
                       
                      <span
                        class="entity-status__label"
                        data-v-bfeabd48=""
                      >
                        Online
                      </span>
                    </div>
                  </td>
                  <td
                    data-v-bfeabd48=""
                  >
                    zone-1
                  </td>
                </tr>
              </tbody>
            </table>
             
            <div
              class="pagination"
            >
              <!---->
               
              <!---->
            </div>
          </div>
           
          <!---->
           
          <!---->
        </div>
         
        <!---->
         
        <!---->
      </div>
       
      <div
        class="tab-container"
        data-testid="tab-container"
      >
        <header
          class="tab__header"
        >
          <h3>
            
          Zone Ingress: 
        
          </h3>
        </header>
         
        <div
          class="tab__content-container"
        >
          <div
            class="k-tabs"
            data-v-a38b664a=""
          >
            <ul
              aria-label="ktabs"
              data-v-a38b664a=""
              role="tablist"
            >
              <li
                aria-controls="panel-0"
                aria-selected="true"
                class="tab-item active"
                data-v-a38b664a=""
                id="overview-tab"
                role="tab"
                tabindex="0"
              >
                <a
                  class="tab-link"
                  data-v-a38b664a=""
                >
                  Overview
                </a>
              </li>
              <li
                aria-controls="panel-1"
                aria-selected="false"
                class="tab-item"
                data-v-a38b664a=""
                id="insights-tab"
                role="tab"
                tabindex="0"
              >
                <a
                  class="tab-link"
                  data-v-a38b664a=""
                >
                  Zone Ingress Insights
                </a>
              </li>
            </ul>
            <div
              aria-labelledby="overview-tab"
              class="tab-container"
              data-v-a38b664a=""
              id="panel-0"
              role="tabpanel"
              tabindex="0"
            >
              <div
                data-v-a38b664a=""
              >
                <!---->
                 
                <div
                  class="empty-state-wrapper"
                  data-v-567e03f2=""
                >
                  <div
                    class="empty-state-title"
                    data-v-567e03f2=""
                  >
                    <!---->
                    <h5
                      data-v-567e03f2=""
                    >
                      <div
                        class="card-icon mb-3"
                        data-v-567e03f2=""
                      >
                        <svg
                          class="kong-icon kong-icon-spinner"
                          data-v-567e03f2=""
                          data-v-a7c9c1d2=""
                          height="42"
                          role="img"
                          viewBox="0 0 16 16"
                          width="42"
                        >
                          <title
                            data-v-a7c9c1d2=""
                          >
                            spinner
                          </title>
                          <g
                            data-v-a7c9c1d2=""
                          >
                            <path
                              d="M13.3154759 4.60336552c.6394482-.63889655.6394482-1.67503449 0-2.31393104-.6388966-.63889655-1.6744828-.63889655-2.3133793 0-.6394483.63889655-.6394483 1.67503449 0 2.31393104.6388965.63889655 1.6744827.63889655 2.3133793 0"
                              data-v-a7c9c1d2=""
                              fill="rgba(0, 0, 0, 0.1)"
                              opacity=".1"
                            />
                            <path
                              d="M13.9609931 6.17246897c-.9037241 0-1.6364138.73268965-1.6364138 1.63641379 0 .90317241.7326897 1.63586207 1.6364138 1.63586207.9037241 0 1.6358621-.73268966 1.6358621-1.63586207 0-.90372414-.732138-1.63641379-1.6358621-1.63641379"
                              data-v-a7c9c1d2=""
                              fill="rgba(0, 0, 0, 0.1)"
                              opacity=".25"
                            />
                            <path
                              d="M10.9938207 11.0105379c-.6388966.6388966-.6388966 1.6750345 0 2.3139311.6388965.6388965 1.6750345.6388965 2.313931 0 .6388966-.6388966.6388966-1.6750345 0-2.3139311-.6388965-.6388965-1.6750345-.6388965-2.313931 0"
                              data-v-a7c9c1d2=""
                              fill="rgba(0, 0, 0, 0.1)"
                              opacity=".35"
                            />
                            <path
                              d="M7.79988966 12.3247448c-.90372414 0-1.63586207.7326897-1.63586207 1.6364138 0 .9031724.73213793 1.6358621 1.63586207 1.6358621.90372413 0 1.63641379-.7326897 1.63641379-1.6358621 0-.9037241-.73268966-1.6364138-1.63641379-1.6364138"
                              data-v-a7c9c1d2=""
                              fill="rgba(0, 0, 0, 0.1)"
                              opacity=".5"
                            />
                            <path
                              d="M2.28706207 11.004469c-.63889655.6388965-.63889655 1.6750344 0 2.313931.63889655.6388966 1.67503448.6388966 2.31393103 0 .63889656-.6388966.63889656-1.6750345 0-2.313931-.63889655-.6388966-1.67503448-.6388966-2.31393103 0"
                              data-v-a7c9c1d2=""
                              fill="rgba(0, 0, 0, 0.1)"
                              opacity=".65"
                            />
                            <path
                              d="M3.27244138 7.80888276c0-.90372414-.73268966-1.63641379-1.63641379-1.63641379-.90372414 0-1.63586207.73268965-1.63586207 1.63641379 0 .90317241.73213793 1.63586207 1.63586207 1.63586207.90372413 0 1.63641379-.73268966 1.63641379-1.63586207"
                              data-v-a7c9c1d2=""
                              fill="rgba(0, 0, 0, 0.1)"
                              opacity=".75"
                            />
                            <path
                              d="M4.59271724 4.60943448c.63889655-.63889655.63889655-1.67503448 0-2.31393103-.63889655-.63889655-1.67503448-.63889655-2.31393103 0-.63889655.63889655-.63889655 1.67503448 0 2.31393103.63889655.63889655 1.67503448.63889655 2.31393103 0"
                              data-v-a7c9c1d2=""
                              fill="rgba(0, 0, 0, 0.1)"
                              opacity=".85"
                            />
                            <path
                              d="M6.16402759 1.6361931c0 .90372414.73213793 1.6364138 1.63586207 1.6364138.90372413 0 1.63641379-.73268966 1.63641379-1.6364138 0-.90372413-.73268966-1.63641379-1.63641379-1.63641379-.90372414 0-1.63586207.73268966-1.63586207 1.63641379z"
                              data-v-a7c9c1d2=""
                              fill="rgba(0, 0, 0, 0.1)"
                              opacity=".95"
                            />
                          </g>
                        </svg>
                      </div>
                      
      Data Loading...
    
                    </h5>
                  </div>
                  <div
                    class="empty-state-content"
                    data-v-567e03f2=""
                  >
                    <p
                      data-v-567e03f2=""
                    />
                    <p
                      data-v-567e03f2=""
                    >
                      <!---->
                    </p>
                  </div>
                </div>
                 
                <!---->
                 
                <!---->
              </div>
            </div>
            <div
              aria-labelledby="insights-tab"
              class="tab-container"
              data-v-a38b664a=""
              id="panel-1"
              role="tabpanel"
              tabindex="0"
            >
              <!---->
            </div>
          </div>
           
          <div>
            <!---->
             
            <!---->
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ZoneIngresses.vue renders snapshot when no multizone 1`] = `
<div>
  <div
    class="zoneingresses"
  >
    <div
      class="empty-state-wrapper"
      data-v-567e03f2=""
    >
      <div
        class="empty-state-title"
        data-v-567e03f2=""
      >
        <!---->
        <h5
          data-v-567e03f2=""
        >
          <svg
            class="kong-icon kong-icon--centered kong-icon-dangerCircle"
            data-v-567e03f2=""
            data-v-a7c9c1d2=""
            height="64"
            role="img"
            viewBox="0 0 16 16"
            width="64"
          >
            <title
              data-v-a7c9c1d2=""
            >
              dangerCircle
            </title>
            <g
              data-v-a7c9c1d2=""
            >
              <path
                d="M0,8a8,8 0 1,0 16,0a8,8 0 1,0 -16,0"
                data-v-a7c9c1d2=""
                fill="#BFBFBF"
              />
              <path
                d="M9 11v2H7v-2h2zm0-8v6.5H7V3h2z"
                data-v-a7c9c1d2=""
                fill="#FFF"
                id="preserveColor"
              />
            </g>
          </svg>
          
    Kuma is running in Standalone mode.
  
        </h5>
      </div>
      <div
        class="empty-state-content"
        data-v-567e03f2=""
      >
        <p
          data-v-567e03f2=""
        >
          <p
            data-v-567e03f2=""
          >
            
      To access this page, you must be running in 
            <strong
              data-v-567e03f2=""
            >
              Multi-Zone
            </strong>
             mode.
    
          </p>
        </p>
        <p
          data-v-567e03f2=""
        >
          <a
            class="k-button primary"
            data-v-38fa8ecf=""
            data-v-567e03f2=""
            href="https://kuma.io/docs/0.6.0/documentation/deployments/"
            target="_blank"
            type="button"
          >
            
      Learn More
    
          </a>
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`ZoneIngresses.vue renders zoneingress insights 1`] = `
<div
  class="tab-container"
  data-testid="tab-container"
>
  <header
    class="tab__header"
  >
    <h3>
      
          Zone Ingress: zone-1
        
    </h3>
  </header>
   
  <div
    class="tab__content-container"
  >
    <div
      class="k-tabs"
      data-v-a38b664a=""
    >
      <ul
        aria-label="ktabs"
        data-v-a38b664a=""
        role="tablist"
      >
        <li
          aria-controls="panel-0"
          aria-selected="false"
          class="tab-item"
          data-v-a38b664a=""
          id="overview-tab"
          role="tab"
          tabindex="0"
        >
          <a
            class="tab-link"
            data-v-a38b664a=""
          >
            Overview
          </a>
        </li>
        <li
          aria-controls="panel-1"
          aria-selected="true"
          class="tab-item active"
          data-v-a38b664a=""
          id="insights-tab"
          role="tab"
          tabindex="0"
        >
          <a
            class="tab-link"
            data-v-a38b664a=""
          >
            Zone Ingress Insights
          </a>
        </li>
      </ul>
      <div
        aria-labelledby="overview-tab"
        class="tab-container"
        data-v-a38b664a=""
        id="panel-0"
        role="tabpanel"
        tabindex="0"
      >
        <!---->
      </div>
      <div
        aria-labelledby="insights-tab"
        class="tab-container"
        data-v-a38b664a=""
        id="panel-1"
        role="tabpanel"
        tabindex="0"
      >
        <div
          class="kong-card noBorder"
          data-v-0029352b=""
          data-v-a38b664a=""
        >
          <!---->
          <!---->
          <div
            class="k-card-body"
            data-v-0029352b=""
          >
            <ul
              class="accordion"
              data-v-0029352b=""
            >
              <li
                class="relative cursor-pointer border-b py-2 px-4"
              >
                <div
                  class="flex py-2"
                >
                  <h4
                    class="text-lg font-medium"
                  >
                    <span
                      class="color-green-400"
                    >
                      

    Connect time: 2021-07-13T08:41:04.556796688Z
  
                    </span>
                     
                    <!---->
                  </h4>
                </div>
                 
                <transition-stub
                  name="accordion"
                >
                  <div>
                    <ul>
                      <div>
                        <div>
                          <h5
                            class="overview-tertiary-title"
                          >
                            
      General Information:
    
                          </h5>
                           
                          <ul>
                            <!---->
                             
                            <li>
                              <strong>
                                Last Connected:
                              </strong>
                              Â 
        on 7/13/2021
      
                            </li>
                             
                            <!---->
                          </ul>
                        </div>
                         
                        <div>
                          <!---->
                        </div>
                      </div>
                    </ul>
                  </div>
                </transition-stub>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
     
    <div>
      <!---->
       
      <!---->
    </div>
  </div>
</div>
`;
